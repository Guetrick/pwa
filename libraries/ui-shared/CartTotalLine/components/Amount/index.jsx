import React from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import styles from './style';

/**
 * The Amount component.
 * @param {Object} props The component props.
 * @return {JSX}
 */
const Amount = ({ amount, currency }) => (
  <div className={styles.amount}>
    {typeof amount === 'string' && <I18n.Text string={amount} />}
    {typeof amount === 'number' && <I18n.Price price={amount} currency={currency} />}
  </div>
);

Amount.propTypes = {
  amount: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]).isRequired,
  currency: PropTypes.string.isRequired,
};

export default Amount;
