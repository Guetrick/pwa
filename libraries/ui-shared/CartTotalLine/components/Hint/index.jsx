import React from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import styles from './style';

/**
 * The Hint component.
 * @param {Object} props The component props.
 * @return {JSX|null}
 */
const Hint = ({ hint }) => (
  hint && (
    <div className={styles.hint}>
      <I18n.Text string={hint} />
    </div>
  )
);

Hint.propTypes = {
  hint: PropTypes.string,
};

Hint.defaultProps = {
  hint: null,
};

export default Hint;
