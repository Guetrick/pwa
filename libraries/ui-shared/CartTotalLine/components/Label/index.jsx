import React from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import styles from './style';

/**
 * The ShippingCostsLabel component.
 * @param {Object} props The component props.
 * @return {JSX|null}
 */
const Label = ({ label, showSeparator }) => {
  if (label === null) {
    return null;
  }

  return (
    <div className={styles.label}>
      <I18n.Text string={label} />
      {showSeparator && ':'}
    </div>
  );
};

Label.propTypes = {
  label: PropTypes.string.isRequired,
  showSeparator: PropTypes.bool,
};

Label.defaultProps = {
  showSeparator: true,
};

export default Label;
