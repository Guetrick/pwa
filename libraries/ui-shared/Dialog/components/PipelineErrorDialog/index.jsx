import React, { Component } from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import BasicDialog from '../BasicDialog';

/**
 * The number of taps required until the dialog switches to/from developer mode.
 */
const requiredTapsToSwitchModes = 10;

/**
 * The timeout between repeated tapping on the message (in ms).
 */
const switchModeTapTimeout = 4000 / requiredTapsToSwitchModes;

/**
 * This component displays a pipeline error dialog.
 * The special behaviour of this message is that the user may tap the message body
 * ten times until it reveals additional information about the error that occured.
 */
class PipelineErrorDialog extends Component {
  static propTypes = {
    actions: PropTypes.arrayOf(PropTypes.shape()).isRequired,
    params: PropTypes.shape({
      entityId: PropTypes.string,
      code: PropTypes.string, // The error code as string.
      message: PropTypes.string.isRequired, // The error message reported by the pipeline.
      messageParams: PropTypes.shape(), // A list of key/value pairs (paramName: paramValue)
      translated: PropTypes.bool,
      pipeline: PropTypes.string.isRequired, // The identifier of the pipeline.
      request: PropTypes.shape.isRequired, // The request as sent to the server.
    }).isRequired,
    message: PropTypes.string,
  };

  static defaultProps = {
    message: '',
  };

  /**
   * Creates the component.
   * @param {Object} props The component props.
   */
  constructor(props) {
    super(props);

    this.tapTimeout = null;
    this.tapCounter = 0;

    this.state = {
      devMode: false, // Indicating whether we are in dev mode.
    };
  }

  /**
   * @return {string} The title based on the current state of the dialog.
   */
  get title() {
    return this.state.devMode
      ? 'Pipeline Error'
      : 'modal.title_error';
  }

  /**
   * @return {JSX} The content component based on the the current state of the dialog.
   */
  get content() {
    return this.state.devMode ? this.renderDevErrorMessage() : this.renderUserErrorMessage();
  }

  /**
   * Clears the tap counter once the
   */
  handleTapTimeout = () => {
    this.tapCounter = 0;
  };

  /**
   * User tapped the message, increase the tap counter and switch view modes if required.
   */
  handleClick = () => {
    this.tapCounter += 1;

    if (this.tapTimeout) {
      // Clear the timeout.
      clearTimeout(this.tapTimeout);
    }

    if (this.tapCounter >= requiredTapsToSwitchModes) {
      // Switch modes and reset the tap counter.
      this.tapCounter = 0;

      this.setState({
        devMode: !this.state.devMode,
      });
    } else {
      this.tapTimeout = setTimeout(this.handleTapTimeout, switchModeTapTimeout);
    }
  };

  /**
   * Renders the error message in developer mode.
   * @returns {JSX}
   */
  renderDevErrorMessage = () => {
    const { params } = this.props;

    return (
      <div aria-hidden>
        <strong>Pipeline:</strong> {params.pipeline}<br />
        { params.entityId && (
          <span>
            <strong>Entity id: </strong> {params.entityId} <br />
          </span>
        )}
        { params.code && (
          <span>
            <strong>Code: </strong> {params.code} <br />
          </span>
        )}
        <strong>Message:</strong> {params.message} <br />
        { params.messageParams && (
          <p>
            <strong>Message Params:</strong><br />
            {JSON.stringify(params.messageParams, null, ' ')}
          </p>
        )}
        { params.translated && (
          <span>
            <strong>Message Translated: </strong> {params.translated.toString()} <br />
          </span>
        )}
        <p>
          <strong>Request Params:</strong><br />
          {JSON.stringify(params.request, null, ' ')}
        </p>
      </div>
    );
  }

  /**
   * Renders the regular error message in user mode.
   * @returns {JSX}
   */
  renderUserErrorMessage = () => {
    const { message = '', params = {} } = this.props;

    return (
      <React.Fragment>
        {params.translated === true && (
          message || params.message || 'modal.body_error'
        )}
        {params.translated !== true && (
          <I18n.Text
            string={message || params.message || 'modal.body_error'}
            params={params.messageParams || {}}
          />
        )}
      </React.Fragment>
    );
  }

  /**
   * Renders the error message depending on the current mode.
   * @return {JSX}
   */
  render() {
    return (
      <BasicDialog title={this.title} actions={this.props.actions}>
        <div aria-hidden onClick={this.handleClick}>
          {this.content}
        </div>
      </BasicDialog>
    );
  }
}

export default PipelineErrorDialog;
