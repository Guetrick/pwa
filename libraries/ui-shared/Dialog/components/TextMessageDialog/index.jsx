import React from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import BasicDialog from '../BasicDialog';

/**
 * Renders a simple dialog with a title and a text message.
 * This is the default dialog if no type has been specified.
 * @param {Object} props The component properties.
 * @returns {JSX} The rendered component.
 */
const TextMessageDialog = ({
  actions, message, title, params,
}) => (
  <BasicDialog title={title} actions={actions}>
    <I18n.Text string={message} params={params} />
  </BasicDialog>
);

TextMessageDialog.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  message: PropTypes.string.isRequired,
  params: I18n.Text.propTypes.params,
  title: PropTypes.string,
};

TextMessageDialog.defaultProps = {
  title: null,
  params: I18n.Text.defaultProps.params,
};

export default TextMessageDialog;
