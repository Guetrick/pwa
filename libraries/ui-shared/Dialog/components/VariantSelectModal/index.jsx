import React from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import { ITEM_PATH } from '@shopgate/pwa-common-commerce/product/constants';
import { bin2hex } from '@shopgate/pwa-common/helpers/data';
import BasicDialog from '../BasicDialog';
import connect from './connector';

/**
 * Reorders the actions for the modal so that the confirmation button will be rendered last.
 * Also attaches a navigation action to the confirmation action.
 * @param {Array} actions The confirm and dismiss actions.
 * @param {string} productId The product id passed through params.
 * @param {string} navigate The navigate action.
 * @return {Array} Reordered and extended actions.
 */
const reorderActions = (actions, { productId }, navigate) => {
  let confirmAction;
  const orderedActions = actions;

  orderedActions.forEach((act, index) => {
    if (act.action.name !== 'onConfirm') {
      return;
    }

    confirmAction = {
      ...act,
      action: () => {
        // Execute default action
        act.action();
        // Navigate to product details page
        if (productId) {
          navigate(`${ITEM_PATH}/${bin2hex(productId)}`);
        }
      },
    };

    orderedActions.splice(index, 1);
  });

  // Push the confirm action last so that we have the button rendered at the bottom right of modal.
  if (confirmAction) {
    orderedActions.push(confirmAction);
  }

  return orderedActions;
};

/**
 * Renders a simple dialog with a title and a text message.
 * @param {Array} actions The modal button actions.
 * @param {string} message The modal message.
 * @param {string} title The modal title.
 * @param {Object} params The modal params.
 * @return {JSX} The rendered component.
 * @constructor
 */
const VariantSelectModal = ({
  actions, message, navigate, title, params,
}) => {
  const parsedActions = reorderActions(actions, params, navigate);

  return (
    <BasicDialog title={title} actions={parsedActions}>
      <I18n.Text string={message} />
    </BasicDialog>
  );
};

VariantSelectModal.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  message: PropTypes.string.isRequired,
  navigate: PropTypes.func.isRequired,
  params: PropTypes.shape(),
  title: BasicDialog.propTypes.title,
};

VariantSelectModal.defaultProps = {
  params: {
    productId: null,
  },
  title: BasicDialog.defaultProps.title,
};

export default connect(VariantSelectModal);

export { VariantSelectModal as UnwrappedVariantSelectModal };
