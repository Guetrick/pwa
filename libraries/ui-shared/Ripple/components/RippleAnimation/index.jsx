import React, { useMemo, memo } from 'react';
import PropTypes from 'prop-types';
import { Transition } from 'react-transition-group';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';
import style from '../../style';

const transitionStyles = {
  entering: {
    transform: 'translate3d(-50%, -50%, 0) scale3d(1, 1, 1)',
    opacity: 0,
  },
  entered: {
    transform: 'translate3d(-50%, -50%, 0) scale3d(1, 1, 1)',
    opacity: 0,
  },
};

/**
 * The ripple animation component.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
function RippleAnimation(props) {
  const {
    color, duration, onComplete, size, x, y,
  } = props;

  const defaultStyles = useMemo(() => ({
    backgroundColor: color,
    height: size,
    width: size,
    transform: 'translate3d(-50%, -50%, 0) scale3d(0, 0, 1)',
    transition: `opacity ${duration}ms cubic-bezier(0.25, 0.1, 0.25, 1), transform ${duration}ms cubic-bezier(0.25, 0.1, 0.25, 1)`,
    left: x,
    top: y,
    opacity: 0.25,
  }), [color, duration, size, x, y]);

  return (
    <Transition in timeout={duration} onEntered={onComplete}>
      {state => (
        <div
          className={style.ripple}
          style={{
            ...defaultStyles,
            ...transitionStyles[state],
          }}
        />
      )}
    </Transition>
  );
}

RippleAnimation.propTypes = {
  color: PropTypes.string,
  duration: PropTypes.number,
  onComplete: PropTypes.func,
  size: PropTypes.number,
  x: PropTypes.number,
  y: PropTypes.number,
};

RippleAnimation.defaultProps = {
  color: themeConfig.colors.dark,
  duration: 300,
  onComplete() { },
  size: 48,
  x: 0,
  y: 0,
};

export default memo(RippleAnimation);
