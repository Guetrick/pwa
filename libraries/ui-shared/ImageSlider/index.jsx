import React, { Component } from 'react';
import { Swiper } from '@shopgate/pwa-common/components';

/**
 * ImageSlider component that uses the Slider for swiping functionality
 */
class ImageSlider extends Component {
  static propTypes = Swiper.propTypes;

  static defaultProps = Swiper.defaultProps;

  /**
   * This component should only render if children change
   * @param {Object} nextProps updated props
   * @returns {boolean}
   */
  shouldComponentUpdate(nextProps) {
    const nextChildren = React.Children.toArray(nextProps.children);
    const oldChildren = React.Children.toArray(this.props.children);

    if (oldChildren.length !== nextChildren.length || nextProps.disabled !== this.props.disabled) {
      return true;
    }

    return !nextChildren.every((child, index) => child === oldChildren[index]);
  }

  /**
   * Renders the component
   * @returns {JSX}
   */
  render() {
    // No slider if there is only one image
    if (React.Children.count(this.props.children) === 1) {
      return React.Children.toArray(this.props.children).pop();
    }

    // Create a Slider.Item for each image
    const imageSliderItems = [];

    React.Children.forEach(this.props.children, (child, index) => {
      const key = child.key ? `${child.key}_${index}` : index;

      imageSliderItems.push((
        <Swiper.Item key={key}>
          {child}
        </Swiper.Item>
      ));
    });

    return (
      <Swiper
        {...this.props}
        classNames={this.props.classNames}
        rebuildOnUpdate
        maxIndicators={10}
      >
        {imageSliderItems}
      </Swiper>
    );
  }
}

export default ImageSlider;
