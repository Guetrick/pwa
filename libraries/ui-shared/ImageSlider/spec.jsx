/* eslint-disable react/jsx-closing-tag-location,eslint-comments/disable-enable-pair */
import React from 'react';
import { shallow, mount } from 'enzyme';
import Image from '@shopgate/pwa-common/components/Image';
import { Swiper } from '@shopgate/pwa-common/components';
import ImageSlider from './index';

describe('<ImageSlider />', () => {
  it('should not use a slider for only one image', () => {
    const wrapper = mount(<ImageSlider>
      <Image srcmap={['foo/bar']} />
    </ImageSlider>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Swiper.Item).exists()).toBe(false);
  });

  it('should wrap each image into a Swiper.Item', () => {
    const wrapper = shallow(<ImageSlider>
      <Image srcmap={['foo/bar']} />
      <Image srcmap={['bar/foo']} />
    </ImageSlider>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Swiper.Item).length).toBe(2);
    expect(wrapper
      .find(Swiper.Item)
      .at(0)
      .find(Image).length).toBe(1);
    expect(wrapper
      .find(Swiper.Item)
      .at(1)
      .find(Image).length).toBe(1);
  });
});
