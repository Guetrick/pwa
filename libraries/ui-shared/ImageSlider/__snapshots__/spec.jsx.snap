// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageSlider /> should not use a slider for only one image 1`] = `
<ImageSlider
  autoPlay={false}
  className={null}
  classNames={Object {}}
  controls={false}
  indicators={false}
  initialSlide={0}
  interval={3000}
  loop={false}
  maxIndicators={null}
  onSlideChange={[Function]}
  rebuildOnUpdate={false}
  slidesPerView={1}
  snapItems={true}
>
  <Image
    alt={null}
    animating={true}
    backgroundColor="#f2f2f2"
    className=""
    forcePlaceholder={false}
    highestResolutionLoaded={[Function]}
    key=".0"
    onError={null}
    onLoad={null}
    ratio={null}
    resolutions={
      Array [
        Object {
          "blur": 2,
          "height": 50,
          "width": 50,
        },
        Object {
          "height": 440,
          "width": 440,
        },
      ]
    }
    src={null}
    srcmap={
      Array [
        "foo/bar",
      ]
    }
    transition={null}
  >
    <div
      className="css-1oue43v "
    />
  </Image>
</ImageSlider>
`;

exports[`<ImageSlider /> should wrap each image into a Swiper.Item 1`] = `
<Swiper
  autoPlay={false}
  className={null}
  classNames={Object {}}
  controls={false}
  indicators={false}
  initialSlide={0}
  interval={3000}
  loop={false}
  maxIndicators={10}
  onSlideChange={[Function]}
  rebuildOnUpdate={true}
  slidesPerView={1}
  snapItems={true}
>
  <SwiperItem
    className={null}
    key="0"
  >
    <Image
      alt={null}
      animating={true}
      backgroundColor="#f2f2f2"
      className=""
      forcePlaceholder={false}
      highestResolutionLoaded={[Function]}
      onError={null}
      onLoad={null}
      ratio={null}
      resolutions={
        Array [
          Object {
            "blur": 2,
            "height": 50,
            "width": 50,
          },
          Object {
            "height": 440,
            "width": 440,
          },
        ]
      }
      src={null}
      srcmap={
        Array [
          "foo/bar",
        ]
      }
      transition={null}
    />
  </SwiperItem>
  <SwiperItem
    className={null}
    key="1"
  >
    <Image
      alt={null}
      animating={true}
      backgroundColor="#f2f2f2"
      className=""
      forcePlaceholder={false}
      highestResolutionLoaded={[Function]}
      onError={null}
      onLoad={null}
      ratio={null}
      resolutions={
        Array [
          Object {
            "blur": 2,
            "height": 50,
            "width": 50,
          },
          Object {
            "height": 440,
            "width": 440,
          },
        ]
      }
      src={null}
      srcmap={
        Array [
          "bar/foo",
        ]
      }
      transition={null}
    />
  </SwiperItem>
</Swiper>
`;
